oemof_b3/tools/data_processing.py

```diff
@@ -592,23 +709,28 @@ def expand_regions(scalars, regions, where="ALL"):
 
     sc_wo_region = _scalars.loc[scalars["region"] == where, :].copy()
 
-    if sc_wo_region.empty:
-        return sc_with_region
-
-    for region in regions:
-        regionalized = sc_wo_region.copy()
-
-        regionalized["name"] = regionalized.apply(
-            lambda x: "-".join([region, x["carrier"], x["tech"]]), 1
-        )
+    # REGIONALIZATION
+    if not sc_wo_region.empty:
```

Comment:
Why did you change the if statement? Before, both cases where covered. Now, if sc_wo_region is empty, you will run into an error when trying to return sc_with_region.

Reply:
No, I don't because the functionality remains same returning `sc_with_region` in line 737 if no regionalization happens (if statement line 715 - 735). I've tested it with data which has region specified. 

While I was doing that, I caught another error, which results from an empty DataFrame in case of no data with unset name. I made the corresponding changes in commit https://github.com/rl-institut/oemof-B3/pull/281/commits/84b67813e00e2326be37e65c4a98937aede07b8d.

Reply:
Ah, you are right, sorry. I missed line 737.